janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java

```diff
@@ -398,11 +415,28 @@ private WriteResult tryWriteLockOnce(StaticBuffer key, StaticBuffer del, StoreTr
     private WriteResult tryDeleteLockOnce(StaticBuffer key, StaticBuffer col, StoreTransaction txh) {
         Throwable t = null;
         final Timer delTimer = times.getTimer().start();
+        StoreTransaction newTx = null;
         try {
-            final StoreTransaction newTx = overrideTimestamp(txh, delTimer.getStartTime());
+            newTx = overrideTimestamp(txh, delTimer.getStartTime());
+
             store.mutate(key, ImmutableList.of(), Collections.singletonList(col), newTx);
+
+            newTx.commit();
+            newTx = null;
         } catch (BackendException e) {
             t = e;
+        } finally {
+            if (newTx != null) {
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Transaction is still open! Rolling back", new Throwable());
+                    }
+
+                    newTx.rollback();
+                } catch (Throwable excp) {
+                    log.debug("Transaction rollback failed", excp);
+                }
+            }
```

Comment:
We have multiple same `finally` blocks. I propose to move the same code into a separate method. I.e.:
```
private void rollbackIfNotNull(BaseTransaction tx){
    if (tx != null) {
        try {
            if (log.isDebugEnabled()) {
                log.debug("Transaction is still open. Rolling back the transaction: "+tx.toString());
            }
            tx.rollback();
        } catch (Throwable e) {
            //TODO: We need to handle the transaction newTx which potentially may leak into memory
            log.error("Failed to rollback the transaction "+tx.toString()+". The transaction now may be leaked into memory.", e);
        }
    }
}
```

After that we can change all places where you added `finally` like the next:
```
finally {
  rollbackIfNotNull(newTx);
}
```

Reply:
I agree. Will update the patch shortly. Thanks!